### PR TITLE
Add Keycloak support

### PR DESCRIPTION
**Pull request by andreasgrill**
Add support for Keycloak instances through the Keycloak OAuth provider
https://github.com/stevenmaguire/oauth2-keycloak.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -83,6 +84,18 @@ providers:
     client_secret: ''
     options:
       scope: ['users']
+  keycloak:
+    enabled: false
+    authserverl_url: ''
```
Comment (an-ant0ni0) [inline README.md]:
```suggestion
    authserver_url: ''
```

File: README.md [hunk 2]
```diff
@@ -134,6 +147,17 @@ admin:
       client_secret: ''
       options:
         scope: ['users']
+    keycloak:
+      enabled: false
+      authserverl_url: ''
```
Comment (an-ant0ni0) [inline README.md]:
```suggestion
      authserver_url: ''
```

**Review by an-ant0ni0 — CHANGES_REQUESTED** [review]

**Comment by rhukster** [comment]
I manually merged this.  thanks!

**Comment by rhukster** [comment]
BTW, please test this as I had to manually merge and didn't test it.

**Comment by andreasgrill** [comment]
Thanks for merging! 
I've tested the changes from the develop branch with our Keycloak instance and it still works.

**Comment by rhukster** [comment]
Ok thanks! going to release.

**Pull request merged**